pkg/controller/deployment/sync.go

```diff
@@ -534,22 +534,6 @@ func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool,
 	if err != nil {
 		return false, err
 	}
-	// If there is no new replica set matching this deployment and the deployment isn't paused
-	// then there is a new rollout that waits to happen
-	if newRS == nil && !d.Spec.Paused {
-		// Update all active replicas sets to the new deployment size. SetReplicasAnnotations makes
-		// sure that we will update only replica sets that don't have the current size of the deployment.
-		maxSurge := deploymentutil.MaxSurge(*d)
-		for _, rs := range controller.FilterActiveReplicaSets(oldRSs) {
-			if updated := deploymentutil.SetReplicasAnnotations(rs, d.Spec.Replicas, d.Spec.Replicas+maxSurge); updated {
```

Comment:
Note that we already have an e2e test that checks for the existense of these annotations for both the new and old replica sets: https://github.com/kubernetes/kubernetes/blob/189bb74213d044a8d183099cc12faddca9b5945d/test/e2e/deployment.go#L87